docs/source/guides/data.rst

```diff
@@ -0,0 +1,358 @@
+
+.. testsetup:: *
+
+    from pytorch_lightning.core.Lightning import LightningModule
+    from torch.utils.data import IterableDataSet
+    from pytorch_lightning.trainer.trainer import Trainer
+
+.. _data:
+
+#############
+Managing Data
+#############
+
+Continue reading to learn about:
+
+* `<Data Containers in Lightning_>`_
+
+* `Iterate over multiple datasets <Multiple DataSets_>`_
+
+* `Handle sequential data <Sequential Data_>`_
+
+****************************
+Data Containers in Lightning
+****************************
+
+There are a few different data containers used in Lightning:
+
+.. list-table:: Data objects
+   :widths: 20 80
+   :header-rows: 1
+
+   * - Object
+     - Definition
+   * - :class:`~torch.utils.data.Dataset`
+     - The PyTorch :class:`~torch.utils.data.Dataset` represents a map from keys to data samples.
+   * - :class:`~torch.utils.data.DataLoader`
+     - The PyTorch :class:`~torch.utils.data.DataLoader` represents a Python iterable over a DataSet.
+   * - :class:`~pytorch_lightning.core.datamodule.LightningDataModule`
+     - A :class:`~pytorch_lightning.core.datamodule.LightningDataModule` is simply a collection of a training DataLoader, validation DataLoader(s) and test DataLoader(s), along with the matching transforms and data processing/downloads steps required.
+
+Why LightningDataModules
+========================
+
+:class:`~pytorch_lightning.core.datamodule.LightningDataModule` were designed as a way of decoupling data-related hooks from the :class:`~pytorch_lightning.core.lightning.LightningModule` so you can develop dataset agnostic models. DataModules make it easy to hot swap different datasets with your model, so you can test it and benchmark it across domains. It also makes sharing and reusing the exact data splits and transforms across projects possible.
+
+Read :ref:`this <datamodules>` for more details on LightningDataModules.
+
+
+.. _multiple-training-dataloaders:
+
+*****************
+Multiple Datasets
+*****************
+
+There are a few ways to pass multiple Datasets to Lightning:
+
+1. Create a DataLoader that iterates over multiple Datasets under the hood.
+2. In the training loop you can pass multiple DataLoaders as a dict or list/tuple and Lightning
+   will automatically combine the batches from different DataLoaders.
+3. In the validation and test loop you have the option to return multiple DataLoaders,
+   which Lightning will call sequentially.
+
+
+Using LightningDataModule
+=========================
+You can set multiple DataLoaders in your :class:`~pytorch_lightning.core.datamodule.LightningDataModule`, and Lightning will handle the
+combination batch under-the-hood.
```

Comment:
will Lightning handle combining the batches under the hood or the combination of batches sent by multiple DataLoaders? Not clear from the last part of the sentence.